src/input.rs

```diff
@@ -77,7 +77,8 @@ mod tests {
     }
 }
 
-fn _cal_index_first_ver(base_index: Option<usize>, layer_index: usize) -> usize {
+#[cfg(test)]
+fn cal_index_first_ver(base_index: Option<usize>, layer_index: usize) -> usize {
```

Comment:
I think it would look better to move the two methods, `cal_index_first_ver` and `cal_index_with_bit_op`, between the line 48 and 49. It should require adjustments regarding paths of variables and methods as well as the declaration of `num_traits::ToPrimitive` inside the `tests`.

Reply:
Thank you for the suggestion! I've made the changes as you recommended and adjusted the paths. Let me know if there's anything else!